src/classes/UTIL_String.cls

```diff
@@ -79,5 +79,29 @@ public class UTIL_String {
 
         return target.replaceAll('[^A-Za-z0-9]', '');
 
-    }    
+    }
+
+    /**
+    * @description Builds an internal URL string from a fully composed URL and also sanitizes an existing internal URL
+    * @param url The url to be sanitized
+    * @return String - The new sanitized internal url. If the fully qualified url is invalid, the method returns null.
+    */
+    public static String getInternalUrlPath(String url) {
+        try {
+
+            String internalUrl = '';
+
+            if (url.startsWith('/')) {
+                internalUrl = '/' + url.replaceFirst('/+', '');
+            } else {
+                internalUrl = new Url(url).getPath();
+            }
+
+            return internalUrl;
+
+        } catch(Exception pokemon) {
```

Comment:
I think we generally assign caught exceptions to a variable named "e" or "ex" - should this Exception be named similarly for consistency?

Reply:
As much as it pains me, we should probably change this to something more generic :)

Reply:
(But at the same time, yes - when it comes to exceptions, you DO have to have catch them all)

Reply:
fixed! Ironically did that out of habit!